File: app/src/main/java/org/deiverbum/app/data/entity/LiturgyWithTime.java
package org.deiverbum.app.data.entity;

import androidx.room.Embedded;
import androidx.room.Relation;

import org.deiverbum.app.model.Liturgy;
import org.deiverbum.app.model.LiturgyTime;

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2022.2
 */
public class LiturgyWithTime {
    @Embedded
    public LiturgyEntity liturgyEntity;

    @Relation(
            parentColumn = "timeFK",
            entityColumn = "timeID",
            entity = LiturgiaTiempoEntity.class
    )
    public LiturgiaTiempoEntity tiempo;

    public Liturgy getDomainModel() {
        Liturgy dm=new Liturgy();
        dm.setLiturgyID(liturgyEntity.liturgiaId);
        dm.setLiturgiaId(liturgyEntity.getLiturgiaId());
        dm.setSemana(liturgyEntity.getSemana());
        dm.setDia(liturgyEntity.getDia());
        dm.setColorId(liturgyEntity.getColorFK());
        dm.setNombre(liturgyEntity.getNombre());
        LiturgyTime t=new LiturgyTime();
        t.setTiempoId(tiempo.getTiempoId());
        t.setTiempo(tiempo.getTiempo());
        t.setLiturgyName(tiempo.getLiturgyName());
        dm.setLiturgiaTiempo(t);
        return dm;
    }


}

File: app/src/main/java/org/deiverbum/app/core/database/model/relation/LiturgyWithTime.kt
package org.deiverbum.app.core.database.model.relation

import androidx.room.Embedded
import androidx.room.Relation
import org.deiverbum.app.core.database.model.entity.LiturgyEntity
import org.deiverbum.app.core.database.model.entity.LiturgyTimeEntity
import org.deiverbum.app.core.database.model.entity.asExternalModel
import org.deiverbum.app.core.model.liturgia.Liturgy

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
data class LiturgyWithTime(
    @Embedded
    val liturgiaEntity: LiturgyEntity,

    @Relation(parentColumn = "timeFK", entityColumn = "timeID")
    val liturgyTime: LiturgyTimeEntity
)

fun LiturgyWithTime.asExternalModel() = Liturgy(
    liturgiaEntity.semana,
    liturgiaEntity.dia,
    //liturgiaEntity.colorFK,
    liturgiaEntity.nombre,
    liturgyTime.asExternalModel()
)
